flake.nix

```diff
@@ -1,86 +1,141 @@
-      ghc-version = "927";
-      systemAttrs = flake-utils.lib.eachDefaultSystem (system:
+  outputs = { self, nixpkgs, flake-utils, haskellNix, flake-compat }:
+    flake-utils.lib.eachSystem [
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ]
+      (system:
         let
-          pkgs = nixpkgs.legacyPackages."${system}".extend self.overlay;
-          ghc = pkgs.haskell.packages."ghc${ghc-version}";
-          nativePackages = pkgs.lib.optionals pkgs.stdenv.isDarwin
-            (with pkgs.darwin.apple_sdk.frameworks; [ Cocoa ]);
-
-          unison-env = pkgs.mkShell {
-            packages = let exports = self.packages."${system}";
-            in with pkgs;
-            [
-              exports.stack
-              exports.hls
-              exports.ormolu
-              exports.ghc
-              pkg-config
-              zlib
-              glibcLocales
-            ] ++ nativePackages;
-            # workaround for https://gitlab.haskell.org/ghc/ghc/-/issues/11042
-            shellHook = ''
-              export LD_LIBRARY_PATH=${pkgs.zlib}/lib:$LD_LIBRARY_PATH
-            '';
+          overlays = [
+            haskellNix.overlay
+            (final: prev: {
+              unison-project = with prev.lib.strings;
+                let
+                  cleanSource = pth:
+                    let
+                      src' = prev.lib.cleanSourceWith {
+                        filter = filt;
+                        src = pth;
+                      };
+                      filt = path: type:
+                        let
+                          bn = baseNameOf path;
+                          isHiddenFile = hasPrefix "." bn;
+                          isFlakeLock = bn == "flake.lock";
+                          isNix = hasSuffix ".nix" bn;
+                        in
+                        !isHiddenFile && !isFlakeLock && !isNix;
+                    in
+                    src';
+                in
+                final.haskell-nix.project' {
+                  src = cleanSource ./.;
+                  projectFileName = "stack.yaml";
+                  modules = [
+                    # enable profiling
+                    {
+                      enableLibraryProfiling = true;
+                      profilingDetail = "none";
+                    }
+                    # remove buggy build tool dependencies
+                    ({ lib, ... }: {
+                      # this component has the build tool
+                      # `unison-cli:unison` and somehow haskell.nix
+                      # decides to add some file sharing package
+                      # `unison` as a build-tool dependency.
+                      packages.unison-cli.components.exes.cli-integration-tests.build-tools = lib.mkForce [ ];
+                    })
+                  ];
+                  branchMap = {
+                    "https://github.com/unisonweb/configurator.git"."e47e9e9fe1f576f8c835183b9def52d73c01327a" =
```

Comment:
Ok so we have to remember to update these; haskell.nix doesn't automatically find them in `stack.yaml`

Reply:
They are automatically found in `stack.yaml`, but if the commit we pinned in our `stack.yaml` isn't in the history of the default branch then we have to specify what branch it will be found in.

Reply:
Ah I see. We can change the default branch or merge these into the default branch if that helps; I guess I'd lean towards the former but open to whatever.